Add an introduction in the readme, remove/update it (#1084)
camptocamp/c2cciutils: README.md

# C2C CI utils

## Publishing

The main goals of C2C CI utils is to offer the commands and the workflows to have the following project structure:

> Have stabilization branch named by default `<major>.<minor>`.

Reviewer: I guess we can have multiple stabilization branches?
```suggestion
Have stabilization branches named by default `<major>.<minor>`.
```

> Have od the release named by default `<major>.<minor>-<patch>`.

Reviewer: Not sure about "od" - maybe a typo?

With C2C CI utils you can publish a python package and a Docker image from the same repository.

The default publishing are:

> - Push on the `<major>.<minor>` branch will publishing Docker images (the version in the last line af the

Reviewer: ```suggestion
- Push on the `<major>.<minor>` branch will publish Docker images (the version in the last line of the
```

  `SECURITY.md` of the `master` branch will be published also using the `latest` tag).
- Create the tag `<major>.<minor>.<patch>` will publish the Docker images, and the Python package.
- Push on a feature branch (whatever other name) will publish the Docker images.
- Delete a feature branch will delete the Docker images.
- Push on the `master` branch will publish the Docker images with the master tag (Publishing a python package is also possible).

The Docker images are published on Docker Hub and GitHub Container Registry.

> .. Note::

Reviewer: I think there is a formatting problem here?

Author: Fixed, thanks :-)

    Dry run publish: `GITHUB_REF=... c2cciutils-publish --dry-run ...`

## Changelog

> Then we create a tag by default with the `changlog` workflow a release is created on GitHub, a changelog is

Reviewer: ```suggestion
When we create a tag by default with the `changelog` workflow a release is created on GitHub, a changelog is
```

> generated add added to the release.

Reviewer: ```suggestion
generated and added to the release.
```

## Security

The security is managed by the `c2cciutils-audit` command with Snyk, it will audit the dependencies of the project on every
stabilization branches, if possible a pull request is created automatically to update the dependencies.

When we publish a Docker image the generated image is monitored by Snyk, this mean that Snyk will search
for all the dependencies and send the list to the Snyk web site to be monitored.
We also do a test of the image and log the result (This will never failed the build).

## Checks

C2C CI utils will no mere provide a tool to do a check of the project, this is replaced by `pre-commit`,
a base configuration is provided in the example project.

## Pull request checks

A workflow is provided to run the checks on the pull requests, it will run the `c2cciutils-pr-checks` command.

- Check that the commit message and the pull request title start with a capital letter.
- Check that there aren't any spelling issue in the commit message and in the pull request title.

> - Add a message to the pull request to the JIRA issue if the pull request branch name start with

Author: For me, it's clear, I spear about this kind of messages: https://github.com/camptocamp/tilecloud/pull/1548#issuecomment-1609391111

> - Add a message to the pull request to the JIRA issue if the pull request branch name start with

Reviewer: OK I see, then I would rather write something like:
```suggestion
- Add a message to the pull request with the JIRA issue if the pull request branch name start with
```

  `[a-zA-Z]+-[0-9]+-` or end with `-[a-zA-Z]+-[0-9]+`.

## Dependencies

In the example project there is a basic Renovate configuration, it will update the dependencies of the project.
There is also a workflow to add a review on the Renovate pull requests to make the auto merge working on
repository that required a review.

## Backports

A workflow is provided to backport the pull requests on the stabilization branches, it will be triggered by
adding a label named `backport <destination_branch>` on the pull request.

## Old workflows

GitHub will retain all the old workflows, so we need to delete them, the `delete-old-workflows-run`
workflow will delete the workflows older than 500 days.

## Workflows

> It make easier to place the following workflows:

Reviewer: The sentence is not clear to me

  the secrets exists in the Camptocamp organization but not shared on all project, then you should add
  your project to the shared list.

> ## Use locally, in the projects that use c2cciutils

Author: Back to the previous content without the deprecated `c2cciutils-checks` for the new project there is the relay shirt section `New project`...
